e2e/react-core/src/react-module-federation.test.ts

```diff
@@ -138,6 +140,164 @@ describe('React Module Federation', () => {
+              </li>
+            </ul>
+            <Routes>
+              <Route path="/" element={<NxWelcome title="home" />} />
+
+              <Route path="/About" element={<About />} />
+            </Routes>
+          </React.Suspense>
+        );
+      }
+
+      export default App;
+      `;
+    });
+
+    // Update remote to use the lib
+    updateFile(`${remote}/src/app/app.tsx`, () => {
+      return `
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+        import styles from './app.module.css';
+        import { version } from '@acme/${lib}';
+
+        import NxWelcome from './nx-welcome';
+
+        export function App() {
+          return (
+            
+            <div className='remote'>
+              Lib version: { version }
+              <NxWelcome title="${remote}" />
+            </div>
+          );
+        }
+
+        export default App;
+      `;
+    });
+
+    // configure the host only use version 0.0.1 of the lib
+    updateFile(`${shell}/module-federation.config.js`, () => {
+      return `
+      module.exports = {
+        name: '${shell}',
+        remotes: ['${remote}'],
+        shared: (libraryName, libraryConfig) =>  {
+          if(libraryName === '@acme/${lib}') {
+            return {
+              ...libraryConfig,
+              requiredVersion: '0.0.1'
+            }
+          }
+        }
+      };      
+      `;
+    });
+
+    const buildOutput = runCLI(`run ${shell}:build:development`);
+    expect(buildOutput).toContain('Successfully ran target build');
+
+    let pattern = new RegExp(`register\\("@acme/${lib}",\\s*"0\\.0\\.1"`, 'g');
+    const mainEntry = readFile(`dist/${shell}/main.js`);
+    expect(mainEntry).toMatch(pattern);
+
+    // update lib to version 0.0.2
+    updateFile(
+      `${lib}/src/lib/${lib}.ts`,
+      `
+      export const version = '0.0.2';
+      `
+    );
+
+    updateJson(`${lib}/package.json`, (json) => {
+      return {
+        ...json,
+        version: '0.0.2',
+      };
+    });
+
+    // configure the remote to use version 0.0.2 of the lib
+    updateFile(`${remote}/module-federation.config.js`, () => {
+      return `
+      module.exports = {
+        name: '${remote}',
+        exposes: {
+          './Module': './src/remote-entry.ts',
+        },
+        shared: (libraryName, libraryConfig) =>  {
+          if(libraryName === '@acme/${lib}') {
+            return {
+              ...libraryConfig,
+              requiredVersion: '0.0.2'
+            }
+          }
+        }
+      };      
+      `;
+    });
+
+    const libbuildOutput = runCLI(`run ${remote}:build:production`);
+    expect(libbuildOutput).toContain('Successfully ran target build');
+
+    // read the remoteEntry.js file and check that the version is 0.0.2
+    const remoteEntry = readFile(`dist/${remote}/remoteEntry.js`);
+    pattern = new RegExp(`register\\("@acme/${lib}",\\s*"0\\.0\\.2"`, 'g');
+    expect(remoteEntry).toMatch(pattern);
```

Comment:
Would be better to run check the apps using a e2e suite, so there are better guarantees that the apps actually do run and load the right versions. i.e. when the remote is loaded from host, it serves its own version.